### DISCUSSION TITLE
How to get all asins

### DISCUSSION THREAD

**Discussion by barshopen** [amzn/selling-partner-api-models#3225] [General]
I'm trying to get a list of all my asins.
I'm using `GET_MERCHANT_LISTINGS_ALL_DATA` report type to retrieve a list of all my asins. 
When trying to find a specific asin from the report - I could not find it :( 
This asin is for an inactive product. Also when trying to use `GET_MERCHANT_LISTINGS_INACTIVE_DATA` - I can't find this asin in any way.
Here's the node.js code I use to create the report.
```
async createReport(credentials, accessToken) {
    const method = “POST”;
    const uri = `/reports/2021-06-30/reports`;
    const lastUpdateDate = moment().subtract(2, “years”).add(1, “days”);
    const body = {
      marketplaceIds: this.marketplaceIds,
      reportType: “GET_MERCHANT_LISTINGS_ALL_DATA”,
      dataStartTime: lastUpdateDate.toISOString(),
      dataEndTime: moment().toISOString()
    };
    let res = await this.executeApiRequestV2(credentials, accessToken, method, uri, {}, body);
    console.log(`createReport `, res);
    return res;
  }
```
Is this a bug? Is there something that I miss? How can I get **ALL** my asins?
Thanks

**Comment by iamricks** [comment]
Try sending it without a `dataStartTime` and `dataEndTime`
